### PR TITLE
[RFR] Architecture and firsts unit-tests

### PR DESCRIPTION
**Pull request by Kmaschta**

### CODE REVIEW

File: test/webhookTest.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    describe('fixer', function() {
+        it('should be called every time', function *() {
+            yield webhook(parser, fixer, commiter)(next);
+            assert(fixer.called);
+        });
+    });
+
+    describe('commiter', function() {
+        it('should be called if fixer\'s content is not null', function *() {
+            fixer = sinon.spy(function() { return 42; });
```
Comment (RobinBressan) [inline test/webhookTest.js]: better to use `sinon.stub().returns(42)`

File: config/default.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "port": 3000,
+    "pretty": true,
+    "defaultResponse": {"response": "OK"}
```
Comment (RobinBressan) [inline config/default.json]: Github should not change its spec tomorrow, you can remove this from conf.

File: lib/webhook.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        var request = this.request;
+
+        // console.log(request.method + ' ' + request.href);
+        // console.log(request.headers);
+        // console.log(request.body);
+
+        var parsedContent = parser(request);
+        var fixedContent = fixer(parsedContent);
+
+        this.body = config.defaultResponse;
```
Comment (RobinBressan) [inline lib/webhook.js]: to update according to the last comment

File: server.js [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
-app.use(function* () {
-	var request = this.request;
+// Request parser
+app.use(bodyParser());
 
-	console.log(request.method + ' ' + request.href);
-	console.log(request.headers);
-	console.log(request.body);
+// Json Response renderer
+app.use(json({pretty: config.pretty, param: 'pretty'}));
```
Comment (RobinBressan) [inline server.js]: Are you sure this is relevant?

File: server.js [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
 
-	this.body = 'Hello world !';
-});
+// Sedbot webhook
```
Comment (RobinBressan) [inline server.js]: Not need for such comment as your middleware is already named `webhook`

File: test/webhookTest.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        it('should be called every time', function *() {
+            yield webhook(parser, fixer, commiter)(next);
+            assert(fixer.called);
+        });
+    });
+
+    describe('commiter', function() {
+        it('should be called if fixer\'s content is not null', function *() {
+            fixer = sinon.stub().returns(42);
+            yield webhook(parser, fixer, commiter)(next);
```
Comment (RobinBressan) [inline test/webhookTest.js]: for the next PRs you can document yourself on the lib `co` and its method `wrap` to use a give a custom `this` to your webhook and therefore mock the request.

Comment (Kmaschta) [inline test/webhookTest.js]: I'll do this, thanks !

File: server.js (hunk 4 above)
Comment (Kmaschta) [inline server.js]: I need it to retrieve Github request body, otherwise `this.body` is empty.

Comment (RobinBressan) [inline server.js]: So why do you use the `bodyParser`?

Comment (Kmaschta) [inline server.js]: Finally, I remove the json middleware and its dependency.

**Pull request merged**